### PR TITLE
always clear the connectTimeout

### PR DESCRIPTION
**Pull request by mcollina**
I have seen the close of my processes being severely delayed by this.
I'm not sure how to test this though.

### CODE REVIEW

**Review by ronag — APPROVED** [review]

**Comment by ronag** [comment]
I'm surprised the `client-unref` test doesn't catch this? Maybe that could be improved?

**Comment by mcollina** [comment]
@ronag I've updated this.

**Review by ronag — APPROVED** [review]

File: lib/core/connect.js [hunk 1]
```diff
@@ -79,15 +79,17 @@ class Connector {
...
+        if (callback) {
           const cb = callback
           callback = null
           cb(null, this)
         }
       })
       .on('error', function (err) {
+        clearTimeout(timeout)
+
         if (callback) {
```
Comment (ronag) [inline lib/core/connect.js:93]: Probably enough to just put a `clearTimeout(timeout)` here.

Comment (mcollina) [reply lib/core/connect.js:93]: Why?

Comment (ronag) [reply lib/core/connect.js:93]: The problem is that we don't clear it in the error case. In the non error case it is already cleared.

Comment (ronag) [reply lib/core/connect.js:93]: It's fine the way you did it as well.

**Pull request merged**